src/Uno.UI/UI/Xaml/Controls/TextBox/OverlayTextBoxView/OverlayTextBoxViewExtension.skia.cs

```diff
@@ -0,0 +1,290 @@
+		if (_owner?.TextBox is { } textBox)
+		{
+			_textBoxView?.UpdateProperties(textBox);
+		}
+	}
+
+	public void UpdateSize()
+	{
+		if (_contentElement is null ||
+			_textBoxView is null ||
+			!_textBoxView.IsDisplayed)
+		{
+			return;
+		}
+
+		var width = (int)(_contentElement.ActualWidth - _contentElement.Padding.Horizontal());
+		var height = (int)(_contentElement.ActualHeight - _contentElement.Padding.Vertical());
+
+		if (_lastSize.Width != width || _lastSize.Height != height)
+		{
+			_lastSize = new Size(width, height);
+			_textBoxView.SetSize(_lastSize.Width, _lastSize.Height);
+		}
+	}
+
+	public void UpdatePosition()
+	{
+		if (_contentElement == null ||
+			_textBoxView == null ||
+			!_textBoxView.IsDisplayed)
+		{
+			return;
+		}
+
+		var transformToRoot = _contentElement.TransformToVisual(Windows.UI.Xaml.Window.Current.Content);
+		var point = transformToRoot.TransformPoint(new Point(_contentElement.Padding.Left, _contentElement.Padding.Top));
+		var pointX = (int)point.X;
+		var pointY = (int)point.Y;
+
+		if (_lastPosition.X != pointX || _lastPosition.Y != pointY)
+		{
+			_lastPosition = new Point(pointX, pointY);
+			_textBoxView.SetPosition(pointX, pointY);
+		}
+	}
+
+	public void SetIsPassword(bool isPassword)
+	{
+		//TODO:MZ:
+		//if (_textBoxView is Entry entry)
+		//{
+		//	entry.Visibility = !isPassword;
+		//}
+	}
+
+	public void Select(int start, int length)
+	{
+		if (_owner.TextBox is not { } textBox)
+		{
+			return;
+		}
+
+		EnsureTextBoxView(textBox);
+		if (textBox.FocusState == FocusState.Unfocused)
+		{
+			// Native control can't handle selection until it is part of visual tree.
+			// Use managed selection until then.
+			_selectionStartCache = textBox.Text.Length >= start ? start : textBox.Text.Length;
+			_selectionLengthCache = textBox.Text.Length >= start + length ? length : textBox.Text.Length - start;
+		}
+		else
+		{
+			SetNativeSelectionBounds(start, start + length);
+		}
+	}
+
+	public int GetSelectionStart()
+	{
+		if (_owner.TextBox is not { } textBox)
+		{
+			return 0;
+		}
+
+		return textBox.FocusState == FocusState.Unfocused ?
+			_selectionStartCache ?? 0 :
+			GetNativeSelectionBounds().start;
+	}
+
+	public int GetSelectionLength()
+	{
+		if (_owner.TextBox is not { } textBox)
+		{
+			return 0;
+		}
+
+		if (textBox.FocusState == FocusState.Unfocused)
+		{
+			return _selectionLengthCache ?? 0;
+		}
+		else
+		{
+			var bounds = GetNativeSelectionBounds();
+			return bounds.end - bounds.start;
+		}
+	}
+
+	private void SetNativeSelectionBounds(int start, int end) => _textBoxView?.SetSelectionBounds(start, end);
+
+	private (int start, int end) GetNativeSelectionBounds() => _textBoxView?.GetSelectionBounds() ?? (0, 0);
+
+	private void EnsureTextBoxView(TextBox textBox)
+	{
+		if (_textBoxView is null ||
+			!_textBoxView.IsCompatible(textBox))
+		{
+			var inputText = GetNativeText();
+			_textBoxView = _textBoxViewFactory(textBox);
+			SetNativeText(inputText ?? string.Empty);
```

Comment:
This is very confusing. We call `GetNativeText`, then `SetNativeText`.

How could that make any difference?

Reply:
We try to get the currently displayed native text, if any from the existing native view, before we replace it with a different view. This can happen for example if someone switches a single-line text box to accept return -> on GTK it must switch from GTK Entry to GTK TextView.